File: VBNetCore.Mvvm/Helpers/TriggerActionCommand.vb
Imports System.Windows
Imports System.Windows.Input
Imports Microsoft.Xaml.Behaviors

Namespace VBNetCore.Mvvm.Helpers
    ''' <summary>
    ''' TriggerAction which links events to commands.
    ''' </summary>
    Public Class TriggerActionCommand
        Inherits TriggerAction(Of DependencyObject)

#Region "Dependency Properties"

        ''' <summary>
        ''' Command DP
        ''' </summary>
        Public Shared ReadOnly CommandProperty As DependencyProperty = DependencyProperty.RegisterAttached("Command", GetType(ICommand), GetType(TriggerActionCommand), New FrameworkPropertyMetadata(Nothing))

        ''' <summary>
        ''' Command that should be called when event is triggered.
        ''' </summary>
        Public Property Command As ICommand
            Get
                Return CType(GetValue(CommandProperty), ICommand)
            End Get
            Set(ByVal value As ICommand)
                SetValue(CommandProperty, value)
            End Set
        End Property

        ''' <summary>
        ''' CommandParameter DP
        ''' </summary>
        Public Shared ReadOnly CommandParameterProperty As DependencyProperty = DependencyProperty.RegisterAttached("CommandParameter", GetType(Object), GetType(TriggerActionCommand), New FrameworkPropertyMetadata(Nothing))

        ''' <summary>
        ''' CommandParameter which should be passed with the Command.
        ''' </summary>
        Public Property CommandParameter As Object
            Get
                Return GetValue(CommandParameterProperty)
            End Get
            Set(ByVal value As Object)
                SetValue(CommandParameterProperty, value)
            End Set
        End Property

#End Region

        ''' <summary>
        ''' Invokes the action.
        ''' </summary>
        ''' <paramname="parameter">The parameter to the action. If the action does not require a parameter, the parameter may be set to a null reference.</param>
        Protected Overrides Sub Invoke(ByVal parameter As Object)
            Dim command = Me.Command
            Dim commandParameter = Me.CommandParameter

            'If an aditional command parameter is specified construct a tuple
            If commandParameter IsNot Nothing Then parameter = (parameter, commandParameter)

            If command.CanExecute(parameter) = True Then command.Execute(parameter)
        End Sub

    End Class
End Namespace

File: VBNetCore.Mvvm/Helpers/TriggerActionCommand.cs
using System.Windows;
using System.Windows.Input;
using Microsoft.Xaml.Behaviors;

namespace NetCore.Mvvm.Helpers
{
    /// <summary>
    /// TriggerAction which links events to commands.
    /// </summary>
    public class TriggerActionCommand : TriggerAction<DependencyObject>
    {

        #region Dependency Properties

        /// <summary>
        /// Command DP
        /// </summary>
        public static readonly DependencyProperty CommandProperty = DependencyProperty.RegisterAttached("Command", typeof(ICommand), typeof(TriggerActionCommand), new FrameworkPropertyMetadata((ICommand)null));

        /// <summary>
        /// Command that should be called when event is triggered.
        /// </summary>
        public ICommand Command
        {
            get => (ICommand)GetValue(CommandProperty);
            set => SetValue(CommandProperty, value);
        }

        /// <summary>
        /// CommandParameter DP
        /// </summary>
        public static readonly DependencyProperty CommandParameterProperty = DependencyProperty.RegisterAttached("CommandParameter", typeof(object), typeof(TriggerActionCommand), new FrameworkPropertyMetadata((object)null));

        /// <summary>
        /// CommandParameter which should be passed with the Command.
        /// </summary>
        public object CommandParameter
        {
            get => GetValue(CommandParameterProperty);
            set => SetValue(CommandParameterProperty, value);
        }

        #endregion

        /// <summary>
        /// Invokes the action.
        /// </summary>
        /// <param name="parameter">The parameter to the action. If the action does not require a parameter, the parameter may be set to a null reference.</param>
        protected override void Invoke(object parameter)
        {
            ICommand command = Command;
            object commandParameter = CommandParameter;

            //If an aditional command parameter is specified construct a tuple
            if (commandParameter != null)
                parameter = (parameter, commandParameter);

            if (command?.CanExecute(parameter) == true)
                command.Execute(parameter);
        }

    }
}
